app/services/stat_builder.rb

```diff
@@ -1,33 +1,16 @@
-# typed: true
+# typed: strict
 # frozen_string_literal: true
 
 # Runs queries and builds the data structure for the summary table.
 class StatBuilder
   extend T::Sig
 
-  sig { returns(T::Hash[Collection, Hash]) }
+  sig { returns(T::Hash[Collection, T::Hash[String, Integer]]) }
   def self.build_stats
-    collections_by_id = Collection.all.index_by(&:id)
-
-    by_state.each_with_object({}) do |(id, val), obj|
-      obj[collections_by_id.fetch(id)] = val.merge('total' => total.fetch(id))
-    end
-  end
-
-  sig { returns(T::Hash[Integer, T::Hash[String, Integer]]) }
-  def self.by_state
-    Work.all.group(:collection_id, :state).count.each_with_object({}) do |((id, state), count), obj|
-      obj[id] ||= {}
-      obj[id][state] = count
-    end
-  end
-  private_class_method :by_state
-
-  sig { returns(T::Hash[Integer, Integer]) }
-  def self.total
-    Work.all.group(:collection_id).count.each_with_object({}) do |(id, count), obj|
-      obj[id] = count
+    Collection.all.index_with do |collection|
+      collection.works.group(:state).count.merge(
+        'total' => collection.works.count
```

Comment:
This causes `2N+1` query where before we had 3 queries.  Can you find a solution where there are only 3 queries?

Reply:
@jcoyne You bet.

Reply:
@jcoyne Got it down to 2 total queries. Pushing in a minute.